### PR TITLE
Env instructions in interpreter

### PR DESCRIPTION
**Pull request by Nashtare**
This PR implements some environment instructions on the interpreter side.

### CODE REVIEW

**Comment by Nashtare** [comment]
I'm not too sure for some of them, still familiarizing myself with this section. I also noticed `SELFBALANCE` was missing from the opcode list, is there any reason for that?

File: evm/src/cpu/kernel/interpreter.rs [hunk 1]
```diff
@@ -609,6 +609,32 @@ impl<'a> Interpreter<'a> {
...
+    fn run_address(&mut self) {
+        self.push(
+            self.generation_state.memory.contexts[self.context].segments
+                [Segment::ContextMetadata as usize]
+                .get(ContextMetadata::Address as usize),
+        )
+    }
+
+    fn run_balance(&mut self) {
+        let offset = self.pop().as_usize();
```
Comment (dlubarov) [inline evm/src/cpu/kernel/interpreter.rs:621]: For this one I think we'd need to hash the address (to get the state trie key) and descend the trie based on that key. Maybe put a TODO for now?

(I don't think it's necessarily a requirement for the interpreter to support all opcodes - we use it for more lightweight unit tests, but we could decide to only exercise things like `BALANCE` in integration tests with the "real" zkEVM. So just calling it unsupported could be an option also.)

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs:621]: Probably better to just close this PR for now I think, I believe there's additional tweaks to be done on some others, and it's indeed not a huge priority.

File: evm/src/cpu/kernel/interpreter.rs [hunk 2]
```diff
@@ -659,6 +685,103 @@ impl<'a> Interpreter<'a> {
...
+    }
+
+    fn run_extcodecopy(&mut self) {
+        let dest_offset = self.pop().as_usize();
+        let offset = self.pop().as_usize();
+        let size = self.pop().as_usize();
+        for i in 0..size {
+            let extcode_byte = self.generation_state.memory.mload_general(
+                0,
+                Segment::KernelAccountCode,
```
Comment (dlubarov) [inline evm/src/cpu/kernel/interpreter.rs:740]: I think this should be `self.context` and `Segment::Code`

File: evm/src/cpu/kernel/interpreter.rs (hunk 1 above)
Comment (dlubarov) [reply evm/src/cpu/kernel/interpreter.rs:621]: I think the others appear fine to me, except the couple others that call `get_trie_data`

File: evm/src/cpu/kernel/interpreter.rs (hunk 2 above)
Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs:740]: I guess I got confused by [this line](https://github.com/mir-protocol/plonky2/blob/b721236eb3a9e069d79de3f37dd384c2dafd08e9/evm/src/cpu/kernel/asm/account_code.asm#L130) in the kernel

Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs:740]: Thanks for the explanation William, and yes indeed I had missed the first pop. Following your recommendation I'll leave it unimplemented.

File: evm/src/cpu/kernel/interpreter.rs (hunk 1 above)
Comment (Nashtare) [reply evm/src/cpu/kernel/interpreter.rs:621]: I've left them as todo as you suggested for now, I may get back to them in a follow-up PR

**Review by dlubarov — APPROVED** [review]
LGTM

**Pull request merged**